### PR TITLE
Nullability + Datetime

### PR DESCRIPTION
**Pull request by AndreiKingsley**

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 rootProject.name = "ggdsl"
 include("ggdsl-api")
 include("ggdsl-lets-plot")
-include("ggdsl-echarts")
+//include("ggdsl-echarts")
```
Comment (devcrocod) [inline settings.gradle.kts:4]: try not to disable modules

File: ggdsl-api/src/main/kotlin/org/jetbrains/kotlinx/ggdsl/dsl/column/columnPointer.kt [hunk 2]
```diff
@@ -27,7 +27,7 @@ public  fun <T : Any> columnPointer(id: String): ColumnPointer<T> =
  * @param T the type of the column
  * @receiver the name of the column
  */
-public inline operator fun <reified T : Any> String.invoke(): ColumnPointer<T> =
+public inline operator fun <reified T> String.invoke(): ColumnPointer<T> =
```
Comment (devcrocod) [inline ggdsl-api/src/main/kotlin/org/jetbrains/kotlinx/ggdsl/dsl/column/columnPointer.kt:30]: Shouldn't it be `Any?`?

File: ggdsl-api/src/main/kotlin/org/jetbrains/kotlinx/ggdsl/dsl/scale.kt [hunk 3]
```diff
@@ -64,16 +65,17 @@ public inline fun <reified DomainType : Any> continuousPos(
  * @param transform the transformation of scale
  * @return new [NonPositionalContinuousScale] with given limits
  */
-public inline fun <reified DomainType : Any, reified RangeType : Any> continuous(
-    domainLimits: Pair<DomainType, DomainType>? = null,
-    rangeLimits: Pair<RangeType, RangeType>? = null,
+public inline fun <reified DomainType, reified RangeType> continuous(
+    domainLimits: Pair<DomainType & Any, DomainType & Any>? = null,
+    rangeLimits: Pair<RangeType & Any, RangeType & Any>? = null,
+    nullValue: RangeType? = null,
```
Comment (devcrocod) [inline ggdsl-api/src/main/kotlin/org/jetbrains/kotlinx/ggdsl/dsl/scale.kt:71]: why is `nullValue` not `DomainType`?

File: ggdsl-api/src/main/kotlin/org/jetbrains/kotlinx/ggdsl/dsl/scale.kt [hunk 4]
```diff
@@ -95,11 +100,12 @@ public inline fun <reified DomainType : Any> categoricalPos(categories: List<Dom
  * @param rangeValues [List] defining the range
  * @return new [NonPositionalCategoricalScale] with given limits
  */
-public inline fun <reified DomainType : Any, reified RangeType : Any> categorical(
-    domainCategories: List<DomainType>? = null,
-    rangeValues: List<RangeType>? = null,
+public inline fun <reified DomainType, reified RangeType> categorical(
+    domainCategories: List<DomainType & Any>? = null,
+    rangeValues: List<RangeType & Any>? = null,
+    nullValue: RangeType? = null,
```
Comment (devcrocod) [inline ggdsl-api/src/main/kotlin/org/jetbrains/kotlinx/ggdsl/dsl/scale.kt:106]: why is `nullValue` not `DomainType`?

File: ggdsl-api/src/main/kotlin/org/jetbrains/kotlinx/ggdsl/dsl/scale.kt [hunk 5]
```diff
@@ -109,10 +115,12 @@ public inline fun <reified DomainType : Any, reified RangeType : Any> categorica
  * @param categoriesToValues [List] of pairs of category to corresponding value.
  * @return new [NonPositionalCategoricalScale] with given limits
  */
-public inline fun <reified DomainType : Any, reified RangeType : Any> categorical(
-    categoriesToValues: List<Pair<DomainType, RangeType>>
+public inline fun <reified DomainType, reified RangeType> categorical(
+    categoriesToValues: List<Pair<DomainType & Any, RangeType & Any>>,
+    nullValue: RangeType? = null,
```
Comment (devcrocod) [inline ggdsl-api/src/main/kotlin/org/jetbrains/kotlinx/ggdsl/dsl/scale.kt:120]: why is `nullValue` not `DomainType`?

**Review by devcrocod — CHANGES_REQUESTED** [review]

File: ggdsl-api/src/main/kotlin/org/jetbrains/kotlinx/ggdsl/dsl/scale.kt [hunk 6]
```diff
@@ -82,9 +84,12 @@ public inline fun <reified DomainType : Any, reified RangeType : Any> continuous
  * @param categories [List] defining the domain
  * @return new [PositionalCategoricalScale] with given categories
  */
-public inline fun <reified DomainType : Any> categoricalPos(categories: List<DomainType>? = null)
+public inline fun <reified DomainType> categoricalPos(
+    categories: List<DomainType & Any>? = null,
```
Comment (devcrocod) [inline ggdsl-api/src/main/kotlin/org/jetbrains/kotlinx/ggdsl/dsl/scale.kt:88]: why is there an intersection of `DomainType` and `Any`? I can't specify null in categories.
Why do we need an intersection at all?

**Comment by AndreiKingsley** [comment]
#90
#91

**Review by devcrocod — APPROVED** [review]

**Pull request merged**